### DISCUSSION TITLE
Best Practice for runtime creation of audio graphs? Also looking for advice on handling many raw PCM streams?

### DISCUSSION THREAD

**Discussion by lukeweston1234** [SamiPerttu/fundsp#64] [General]
Hello, I am running into a bit of confusion with how I might spin up new graphs or mixers with new live sound sources. For the time being, I just made a fixed amount of them at compile time, and I am sending samples in via crossbeam for each track. I would like to have a more programmatic approach, where I don't have to worry about dealing with constants in the structs if at all possible. 

Additionally, I had another implementation before this project where I was pulling samples from an iterator, and using a double buffer to update these samples, and this gave me much better performance with large amounts of tracks. I am wondering if there are any examples floating around there that use a technique like this to send samples to an audio graph, as opposed to sending them in via crossbeam/ a ring buffer. Thanks!

**Comment by SamiPerttu** [comment]
Hi! `Sequencer` exists for this purpose. The example `keys` uses it as a dynamic mixer.
